stdlib/libLLVM_jll/src/libLLVM_jll.jl

```diff
@@ -5,41 +5,41 @@
 baremodule libLLVM_jll
 using Base, Libdl
 
-const PATH_list = String[]
-const LIBPATH_list = String[]
-
 export libLLVM
 
 # These get calculated in __init__()
 const PATH = Ref("")
+const PATH_list = String[]
 const LIBPATH = Ref("")
+const LIBPATH_list = String[]
 artifact_dir::String = ""
-libLLVM_handle::Ptr{Cvoid} = C_NULL
 libLLVM_path::String = ""
 
+_libLLVM_dependencies = LazyLibrary[]
+
 if Sys.iswindows()
-    const libLLVM = "$(Base.libllvm_name).dll"
+    const _libLLVM_path = BundledLazyLibraryPath("$(Base.libllvm_name).dll")
 elseif Sys.isapple()
-    const libLLVM = "@rpath/libLLVM.dylib"
+    const _libLLVM_path = BundledLazyLibraryPath("libLLVM.dylib")
 else
-    const libLLVM = "$(Base.libllvm_name).so"
+    const _libLLVM_path = BundledLazyLibraryPath("$(Base.libllvm_name).so")
 end
 
+const libLLVM = LazyLibrary(
+    _libLLVM_path,
+    dependencies=_libLLVM_dependencies,
```

Comment:
```suggestion
    dependencies=[libstdcxx, libgcc_s, libz],
```

I will note that technically there are all these dependencies, and we should capture them, but it's probably not going to hurt us yet, as `libLLVM` is generally either loaded before Julia code gets to run, or not loaded at all.